### PR TITLE
Authentication Added

### PR DESCRIPTION
**Pull request by jeetbhalani15**
## Authentication For Login , Signup & Logout  Widgets
- Added auth functinality for Login
   Login => form validation & Guest Login Feature Added
- Added auth functionality for signup.
   Signup => form validation.
- Added Logout page.
- Will make seprate folder for form validation.

image
image

### CODE REVIEW

File: Zero-Gravity-Store-React/src/Components/Navigation/Navigation.js [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
 import logo from "../../Asserts/Images/hero-logo.png";
-import {Link} from "react-router-dom";
+import { Link } from "react-router-dom";
 import { useState, useEffect } from "react";
-import {  useFliters } from "../../Contexts/Filter-context";
-export  function Navigation() {
-
-  const [search,setSearch]= useState("");
+import { useFliters } from "../../Contexts/Filter-context";
+import { useAuth } from "../../Contexts/Auth-context";
```
Comment (sakshi006) [inline Zero-Gravity-Store-React/src/Components/Navigation/Navigation.js:6]: Just a suggestion, react packages and tools should be imported first, then CSS files and components from other folders.

File: Zero-Gravity-Store-React/src/Pages/AuthWidget/Login/Login.css [hunk 2]
```diff
@@ -122,10 +122,33 @@
...
+  .eye-btn{
+    background: none;
+    border: none;
+    padding: 2rem;
+    /* padding: 2rem; */
+    position: absolute;
+    right: 2%;
+  }
+  .form-err-icon{
+    color: crimson;
```
Comment (sakshi006) [inline Zero-Gravity-Store-React/src/Pages/AuthWidget/Login/Login.css:141]: Use CSS variables for colors

**Review by sakshi006 — APPROVED** [review]
Added few comments.

File: Zero-Gravity-Store-React/src/Pages/AuthWidget/Login/Login.css [hunk 3]
```diff
@@ -122,10 +122,33 @@
...
-    right: 1%;
-    top: 18%;
+    right: 2%;
+    top: 21%;
+  }
+  .eye-btn{
+    background: none;
+    border: none;
+    padding: 2rem;
+    /* padding: 2rem; */
```
Comment (harshitpaliwal95) [inline Zero-Gravity-Store-React/src/Pages/AuthWidget/Login/Login.css:136]: remove commented code

**Review by harshitpaliwal95 — COMMENTED** [review]
code looks good u can merge

File: Zero-Gravity-Store-React/src/Pages/AuthWidget/Login/Login.css (hunk 3 above)
Comment (jeetbhalani15) [reply Zero-Gravity-Store-React/src/Pages/AuthWidget/Login/Login.css:136]: ok you can approve PR if evrything is fine :)

**Pull request merged**